### ISSUE TITLE
Clarification regarding STAT table requirements

### ISSUE THREAD

**Issue by aaronbell** [MicrosoftDocs/typography-issues#808]
I'm hoping to get clarification regarding the requirements for the [STAT table](https://docs.microsoft.com/en-us/typography/opentype/spec/stat). 

In reading through the spec document, I note that there appears to be a difference between the stated requirements of the stat table and the 'spirit' of the purpose of the stat table. 

For example, under 'Alternate Font Family Models', the spec states:
> The 'STAT' accommodates this through axis value tables. An axis value table can provide a string label for a specific axis value on any axis that is relevant for the font or font family. (For special cases, an axis value table can also provide a label for a combination of values from a set of of relevant axes.) These labels, together with the typographic family name (name ID 16), can be used to compose alternate names dynamically to fit the requirements of different font family models.

This indicates to me that it is possible for an application to dynamically generate the axis instances using the stat table. 

However, under 'Axis Value Tables':
> In a variable font, every element of typographic subfamily names for all of the named instances defined in the 'fvar' table **should** be reflected in an axis value table. 

The word 'should' here indicates that it is not absolutely necessary for a stat table to replicate what is in the fvar table. Is this the case?

---

The reason I mention this is that Mac Word's implementation of variable fonts relies on both the fvar table, and stat table, to enumerate the font in the font dropdown. It will take the instances listed in the fvar table, and the names referenced in the stat table to dynamically generate the instance names.

As such, for a font such as Reem Kufi which has a basic stat table:
```
  <STAT>
    <Version value="0x00010001"/>
    <DesignAxisRecordSize value="8"/>
    <!-- DesignAxisCount=1 -->
    <DesignAxisRecord>
      <Axis index="0">
        <AxisTag value="wght"/>
        <AxisNameID value="256"/>  <!-- Weight -->
        <AxisOrdering value="0"/>
      </Axis>
    </DesignAxisRecord>
    <!-- AxisValueCount=0 -->
    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
  </STAT>
```

will render as
<img width="293" alt="127788876-473b302c-266f-4ade-99f2-bd3a39e64512" src="https://user-images.githubusercontent.com/8460297/127901812-564a05ea-ac35-4f9b-9c5c-d24d89765073.png">

The only solution for this issue is to fully enumerate what is present in the fvar table in the STAT table, which would indicate to me that the above **should** should actually be a **must**.

**Comment by davelab6** [comment]
@PeterCon This is important for Google Fonts :)

**Comment by PeterConstable** [comment]
>This indicates to me that it is possible for an application to dynamically generate the axis instances using the stat table.

The STAT spec explicitly discusses dynamic generation of family / sub-family names for alternate family models:

>An application can use these to dynamically compose an alternate family name “Sitka Display” and alternate subfamily name “Bold” to fit the requirements of these alternate font family models.

The primary intent for the STAT table was to allow variable fonts to work in existing applications (of which there are many) that would not understand the possibility of open-ended typographic families, which was an implicit corollary of variable fonts. If implementation of existing platform APIs were updated to utilize STAT data, then existing apps (without updating) calling those APIs could access the named instances in a variable font. It was with that usage in mind that Microsoft, at least, felt that STAT should be a required table.

Of course, that implies an assumption that there would be at least _some_ Axis Value Tables. 

>However, under 'Axis Value Tables':
>
>>In a variable font, every element of typographic subfamily names for all of the named instances defined in the 'fvar' table should be reflected in an axis value table.
>
>The word 'should' here indicates that it is not absolutely necessary for a stat table to replicate what is in the fvar table. Is this the case?

It's five years since this was all fresh in mind, so I can't say for certain why that wording was chosen. I suppose it might have been to give a font developer the option of leaving out certain named instances if they weren't concerned about those being supported in legacy apps.

**Comment by PeterCon** [comment]
I think Word (DWrite, probably) should be more lenient in how it handles instance names, falling back to fvar if the STAT table doesn't have all the data it expects. But I'd still emphasize that the axis value tables really should be included: part of the intent is to give flexibility to apps around UI, and that won't work well if AVTs aren't provided. Word might not be the only app that displays the wrong thing.

**Comment by PeterCon** [comment]
For OT 1.9, the following small addition to the STAT chapter is proposed:

><ins>In a variable font, it is strongly recommended that axis value tables be included for every element of typographic subfamily names for all of the named instances defined in the &#39;fvar&#39; table.</ins>
>
>> <ins>*Note:* Some applications could rely on these axis value tables to compose subfamily names for named instances. Unexpected user interface behavior could result if a complete set of axis value tables is not included in the font.</ins>

**Comment by PeterCon** [comment]
Addressed in [OpenType 1.9](https://docs.microsoft.com/en-us/typography/opentype/spec/). Closing.
